docs/guides/server/containers.adoc

```diff
@@ -54,12 +54,24 @@ The build process includes multiple stages:
 * In the final image, additional configuration options for the hostname and database are set so that you don't need to set them again when running the container.
 * In the entrypoint, the `kc.sh` enables access to all the distribution sub-commands.
 
-To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory:
+To install custom providers, you just need to define a step to include the JAR file(s) into the `/opt/keycloak/providers` directory.
+This step must be placed _after_ the first `FROM` line and before the line that `RUNs` the `build` command, as below: 
```

Comment:
I would not emphasise the first `FROM`. There can be some more stages, see e.g.
https://github.com/keycloak/keycloak/blob/816ed5d3e0f0885a6bb57034be2433a7c7ac78aa/docs/guides/server/containers.adoc?plain=1#L79-L84

Reply:
Thank you. I have incorporated your suggestions.